gremlin-dotnet/src/Gremlin.Net/Driver/ConnectionPoolSettings.cs

```diff
@@ -0,0 +1,55 @@
+#region License
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#endregion
+
+using System;
+
+namespace Gremlin.Net.Driver
+{
+    /// <summary>
+    ///     Holds settings for the <see cref="ConnectionPool"/>.
+    /// </summary>
+    public class ConnectionPoolSettings
+    {
+        private const int DefaultMinPoolSize = 8;
+        private const int DefaultMaxPoolSize = 128;
+        private static readonly TimeSpan DefaultWaitForConnectionTimeout = TimeSpan.FromSeconds(3);
+
+        /// <summary>
+        ///     Gets or sets the minimum size of a connection pool.
```

Comment:
Once we have a pattern for this:

https://issues.apache.org/jira/browse/TINKERPOP-2010

We should do the same for .NET so this nice structured documentation gets published.

Reply:
Yes, that's a good idea. The API docs should be published somewhere. I already did this for the early pre-TinkerPop version of Gremlin.Net: https://florianhockmann.github.io/Gremlin.Net/api/Gremlin.Net.html
We could probably use a similar approach again.

Reply:
I created [TINKERPOP-2018](https://issues.apache.org/jira/browse/TINKERPOP-2018) to track this.